File: 0077-combinations/0077-combinations.cpp
class Solution {
public:
    void f(int n, int k, int s, vector<int>& tmp, vector<vector<int>>* res){
        if(tmp.size() == k){
            res->push_back(tmp);
            return;
        }
        for(int i = s; i <= n; ++i){
            tmp.push_back(i);
            f(n, k, i + 1, tmp, res);
            tmp.pop_back();
        }
    }

    vector<vector<int>> combine(int n, int k) {
        vector<int> tmp;
        vector<vector<int>> ans{};
        f(n, k, 1, tmp, &ans);
        return ans;
    }
};

File: 0077-combinations/0077-combinations.erl
-spec combine(N :: integer(), K :: integer()) -> [[integer()]].
combine(N, K) ->
    bcktrk(N, K, 1, 0, []).

bcktrk(_, K, _, K, Tmp) -> [Tmp];
bcktrk(N, K, S, Len, Tmp) ->
    lists:foldl(
        fun(X, Res) ->
            Res ++ bcktrk(N, K, X + 1, Len + 1, Tmp ++ [X])
        end, [], lists:seq(S, N)
    ).